RN-4.14-fixes: sweeping rns edits (#66990)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

By default, the installation program creates and attaches security groups to control plane and compute machines. The rules associated with the default security groups cannot be modified.

> With {product-title} {product-version}, if you deploy a cluster to an existing Amazon Virtual Private Cloud (VPC), you can apply additional existing AWS security groups to control plane and compute machines. These security groups must be associated with the VPC that you are deploying the cluster to. Applying custom security groups can help you meet the security needs of your organization, in such cases where you must control the incoming or outgoing traffic of these machines.For more information, see xref:../installing/installing_aws/installing-aws-vpc.adoc#installation-aws-vpc-security-groups_installing-aws-vpc[Applying existing AWS security groups to the cluster].

Reviewer: ```suggestion
With {product-title} {product-version}, if you deploy a cluster to an existing Amazon Virtual Private Cloud (VPC), you can apply additional existing AWS security groups to control plane and compute machines. These security groups must be associated with the VPC that you are deploying the cluster to. Applying custom security groups can help you meet the security needs of your organization, in such cases where you must control the incoming or outgoing traffic of these machines. For more information, see xref:../installing/installing_aws/installing-aws-vpc.adoc#installation-aws-vpc-security-groups_installing-aws-vpc[Applying existing AWS security groups to the cluster].
```

With this release, there are several updates to the *Administrator* perspective of the web console. You can now perform the following actions:

* Narrow down the list of resources in a list view or search page with exact search capabilities. This will help when you have similarly named resources and fuzzy search does not narrow down your search.
* Provide direct feedback about features and report a bug by clicking the *Help* button on the toolbar and clicking *Share Feedback* from the list.
* Display and hide tooltips in the YAML editor. The tooltips will persist so you don't have to change it every time you navigate to the page.

> * Configure the web terminal image for all users. For more information, see xref:../web_console/web_terminal/configuring-web-terminal.adoc#configuring-web-terminal[Configuring the web terminal]

Reviewer: ```suggestion
* Configure the web terminal image for all users. For more information, see xref:../web_console/web_terminal/configuring-web-terminal.adoc#configuring-web-terminal[Configuring the web terminal].
```

[id="web-console-dynamic-plugin-enhancements"]
===== Dynamic plugin enhancements

> With this update, you can add custom metric dashboards and extend the cluster's *Overview* page with the `QueryBowser` extension. Additional extension points have been added which allow you to add different types of modals, set the active namespace, provide custom error pages, and set proxy timeouts for your dynamic plugin. For more information, see xref:../web_console/dynamic-plugin/dynamic-plugins-reference.adoc#dynamic-plugin-reference[Dynamic plugin reference] and xref:../web_console/dynamic-plugin/dynamic-plugins-reference.html#dynamic-plugin-api_dynamic-plugins-reference[QueryBrowser in the {product-title} console API].

Reviewer: ```suggestion
With this update, you can add custom metric dashboards and extend the cluster's *Overview* page with the `QueryBowser` extension. Additional extension points have been added which allow you to add different types of modals, set the active namespace, provide custom error pages, and set proxy timeouts for your dynamic plugin. For more information, see xref:../web_console/dynamic-plugin/dynamic-plugins-reference.adoc#dynamic-plugin-reference[Dynamic plugin reference] and xref:../web_console/dynamic-plugin/dynamic-plugins-reference.adoc#dynamic-plugin-api_dynamic-plugins-reference[QueryBrowser in the {product-title} console API].
```

[id="oc-new-app-enhancement"]
==== Enhancement to oc new-app

> A new oc command line interface (CLI) flag, `--import-mode`, has been added to the `oc new-app` command. With this enhancement, users can set the `--import-mode` flag to `Legacy` or `PreserverOriginal`, which provides users the option to create new applications using a single sub-manifest, or all manifests, respectively. For more information, see xref:../applications/creating_applications/creating-applications-using-cli.adoc#setting-the-import-mode[Setting the import mode].

Reviewer: ```suggestion
A new `oc` command line interface (CLI) flag, `--import-mode`, has been added to the `oc new-app` command. With this enhancement, users can set the `--import-mode` flag to `Legacy` or `PreserverOriginal`, which provides users the option to create new applications using a single sub-manifest, or all manifests, respectively. For more information, see xref:../applications/creating_applications/creating-applications-using-cli.adoc#setting-the-import-mode[Setting the import mode].
```

{product-title} now supports configuring {op-system-first} nodes for IBM Secure Execution on {ibmzProductName} and {linuxoneProductName} (s390x architecture).

> For installation instructions, see xref:../installing/installing_ibm_z/installing-ibm-z-kvm.html#installing-rhcos-using-ibm-secure-execution_installing-ibm-z-kvm[Installing {op-system} using IBM Secure Execution].

Reviewer: ```suggestion
For installation instructions, see xref:../installing/installing_ibm_z/installing-ibm-z-kvm.adoc#installing-rhcos-using-ibm-secure-execution_installing-ibm-z-kvm[Installing {op-system} using IBM Secure Execution].
```

[id="ocp-4-14-networking-nmstate-vrf"]
==== Support for VRF specification in NMState's NodeNetworkConfigurationPolicy resource (Technology Preview)

With this update, you can associate a Virtual Routing and Forwarding (VRF) instance with a network interface by using a `NodeNetworkConfigurationPolicy` custom resource. By associating a VRF instance with a network interface, you can support traffic isolation, independent routing decisions, and the logical separation of network resources. This feature is available as a Technology Preview feature. For more information, see xref:../networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.adoc#virt-example-host-vrf_k8s_nmstate-updating-node-network-config[Example: Network interface with a VRF instance node network configuration policy].

[id="ocp-414-broadcom-bcm57504-support"]
==== Support for Broadcom BCM57504 is now GA

> Support for the Broadcom BCM57504 network interface controller is now available for the SR-IOV Network Operator.For more information, see xref:../networking/hardware_networks/about-sriov.adoc#supported-devices_about-sriov[Supported devices].

Reviewer: ```suggestion
Support for the Broadcom BCM57504 network interface controller is now available for the SR-IOV Network Operator. For more information, see xref:../networking/hardware_networks/about-sriov.adoc#supported-devices_about-sriov[Supported devices].
```

[id="ocp-4-14-networking-ovn-kubernetes-secondary-network"]
==== OVN-Kubernetes is available as a secondary network

> With this release, the {openshift-networking} OVN-Kubernetes network plugin allows the configuration of secondary network interfaces for pods. As a secondary network, OVN-Kubernetes supports both layer 2 switched and localnet switched topology networks.For more information about OVN-Kubernetes as a secondary network, see xref:../networking/multiple_networks/configuring-additional-network.adoc#configuration-ovnk-additional-networks_configuring-additional-network[Configuration for an OVN-Kubernetes additional network].

Reviewer: ```suggestion
With this release, the {openshift-networking} OVN-Kubernetes network plugin allows the configuration of secondary network interfaces for pods. As a secondary network, OVN-Kubernetes supports both layer 2 switched and localnet switched topology networks. For more information about OVN-Kubernetes as a secondary network, see xref:../networking/multiple_networks/configuring-additional-network.adoc#configuration-ovnk-additional-networks_configuring-additional-network[Configuration for an OVN-Kubernetes additional network].
```

[id="ocp-4-14-storage-standardized-sts-config"]
==== Standardized STS configuration workflow
{product-title} 4.14 provides a streamlined and standardized procedure to configure Security Token Service (STS) with the AWS Elastic File Storage (EFS) Container Storage Interface (CSI) Driver Operator. For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-aws-efs.adoc#efs-sts_persistent-storage-csi-aws-efs[Configuring AWS EFS CSI Driver Operator with Security Token Service].

[id="ocp-4-14-storage-rwop-access-mode"]
==== Read Write Once Pod access mode (Technology Preview)

> {product-title} 4.14 introduces a new access mode for persistent volumes (PVs) and persistent volume claims (PVCs) called ReadWriteOncePod (RWOP), which can be used only in a single pod on a single node. This is compared to the existing `ReadWriteOnce` access mode where a PV or PVC can be used on a single node by many pods. This feature is supported with Technology Preview status. For more information, see xref:../storage/understanding-persistent-storage.adoc#pv-access-modes_understanding-persistent-storage[Access modes].

Reviewer: I notice that the "`ReadWriteOnce` access mode" is marked up, but the "new access mode [...] called ReadWriteOncePod" is not marked up. It looks like in the [linked-to section](https://66990--docspreview.netlify.app/openshift-enterprise/latest/storage/understanding-persistent-storage#pv-access-modes_understanding-persistent-storage), none of the access modes are marked up there, FWIW. 🤷

[id="ocp-4-14-storage-gcp-filestore-csi-driver"]
==== GCP Filestore storage CSI Driver Operator is generally available

> {product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Google Compute Platform (GCP) Filestore Storage. The GCP Filestore CSI Driver Operator was introduced in {product-title} 4.12 with Technology Preview support. The GCP Filestore CSI Driver Operator is now generally available. For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-google-cloud-file.adoc[Google Compute Platform Filestore CSI Driver Operator].

Reviewer: ```suggestion
{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Google Compute Platform (GCP) Filestore Storage. The GCP Filestore CSI Driver Operator was introduced in {product-title} 4.12 with Technology Preview support. The GCP Filestore CSI Driver Operator is now generally available. For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-google-cloud-file.adoc#persistent-storage-csi-google-cloud-file[Google Compute Platform Filestore CSI Driver Operator].
```

[id="ocp-4-14-storage-automigration-vsphere"]
==== Automatic CSI migration for VMware vSphere
This feature automatically translates in-tree objects to their counterpart CSI representations and should be completely transparent to users. Although storage class referencing to the in-tree storage plug-in will continue working, it is recommended that you switch the default storage class to the CSI storage class.

In {product-title} 4.14, CSI migration for vSphere is enabled by default under all circumstances and requires no action by an administrator.

> However, if you are using vSphere in-tree persistent volumes (PVs) and want to upgrade from {product-title} 4.12 or 4.13 to 4.14, update vSphere vCenter and ESXI host to 7.0 Update 3l or 8.0 Update 2, otherwise the {product-title} upgrade is blocked. If you do not want to update vSphere, you can proceed with an {product-title} upgrade by performing an administrator acknowledgment. However, with using the administrator acknowledgment, known issues can occur. link:https://access.redhat.com/node/7011683[Before proceeding with the administrator acknowledgement, carefully read this knowledge base article].

Reviewer: Nit: "with"? "when"?
```suggestion
However, if you are using vSphere in-tree persistent volumes (PVs) and want to upgrade from {product-title} 4.12 or 4.13 to 4.14, update vSphere vCenter and ESXI host to 7.0 Update 3l or 8.0 Update 2, otherwise the {product-title} upgrade is blocked. If you do not want to update vSphere, you can proceed with an {product-title} upgrade by performing an administrator acknowledgment. However, when using the administrator acknowledgment, known issues can occur. link:https://access.redhat.com/node/7011683[Before proceeding with the administrator acknowledgement, carefully read this knowledge base article].
```

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-migration.adoc[CSI automatic migration].

[id="ocp-4-14-storage-secrets-store-csi-driver-operator"]
==== Secrets Store CSI Driver Operator (Technology Preview)

> The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as an inline ephemeral volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status. For more information about the {secrets-store-driver}, see xref:../storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc[{secrets-store-driver}].

Reviewer: ```suggestion
The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as an inline ephemeral volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. This feature is supported with Technology Preview status. For more information about the {secrets-store-driver}, see xref:../storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc#persistent-storage-csi-secrets-store[{secrets-store-driver}].
```

[id="ocp-4-14-pdb-unhealthy-pod-eviction-policy"]
==== Pod disruption budget (PDB) unhealthy pod eviction policy

> With this release, specifying an unhealthy pod eviction policy for pod disruption budgets (PDBs) is Generally Available in {product-title} and has been removed from the `TechPreviewNoUpgrade` featureset. This helps evict malfunctioning applications during a node drain. For more information, see xref:../nodes/pods/nodes-pods-configuring.adoc#pod-disruption-eviction-policy_nodes-pods-configuring[Specifying the eviction policy for unhealthy pods].

Reviewer: ```suggestion
With this release, specifying an unhealthy pod eviction policy for pod disruption budgets (PDBs) is generally available in {product-title} and has been removed from the `TechPreviewNoUpgrade` feature set. This helps evict malfunctioning applications during a node drain. For more information, see xref:../nodes/pods/nodes-pods-configuring.adoc#pod-disruption-eviction-policy_nodes-pods-configuring[Specifying the eviction policy for unhealthy pods].
```
